test/engine/update.result

```diff
@@ -850,3 +850,305 @@ s:update({1}, {{'=', 'field3', {d = 30, e = 31, f = 32}}})
+---
+- [1, 2, null, 0]
+...
+s:drop()
+---
+...
+-- Ops that must still fail
+format = {}
+---
+...
+format[1] = {name = 'field1', type = 'unsigned'}
+---
+...
+format[2] = {name = 'field2', type = 'unsigned'}
+---
+...
+format[3] = {name = 'field3', type = 'unsigned', is_nullable = true}
+---
+...
+s = box.schema.create_space('test', {format = format})
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2, box.NULL}
+---
+- [1, 2, null]
+...
+s:update({1}, {{'+', 3, 0}})
+---
+- error: 'Argument type in operation ''+'' on field 3 does not match field type: expected
+    a number'
+...
+s:update({1}, {{'-', 3, 0}})
+---
+- error: 'Argument type in operation ''-'' on field 3 does not match field type: expected
+    a number'
+...
+s:update({1}, {{'&', 3, 0}})
+---
+- error: 'Argument type in operation ''&'' on field 3 does not match field type: expected
+    a positive integer'
+...
+s:update({1}, {{'|', 3, 0}})
+---
+- error: 'Argument type in operation ''|'' on field 3 does not match field type: expected
+    a positive integer'
+...
+s:update({1}, {{'^', 3, 0}})
+---
+- error: 'Argument type in operation ''^'' on field 3 does not match field type: expected
+    a positive integer'
+...
+s:update({1}, {{':', 3, 1, 0, '!!'}})
+---
+- error: 'Argument type in operation '':'' on field 3 does not match field type: expected
+    a string'
+...
+s:drop()
+---
+...
+-- Negative field number, fixed field_count
+format = {}
+---
+...
+format[1] = {name = 'field1', type = 'unsigned'}
+---
+...
+format[2] = {name = 'field2', type = 'unsigned'}
+---
+...
+format[3] = {name = 'field3', type = 'unsigned', is_nullable = true}
+---
+...
+format[4] = {name = 'field4', type = 'unsigned', is_nullable = true}
+---
+...
+s = box.schema.create_space('test', {format = format, field_count = 4})
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2, box.NULL, box.NULL}
+---
+- [1, 2, null, null]
+...
+s:update({1}, {{'!', -1, 42}})
+---
+- error: Tuple field count 5 does not match space field count 4
+...
+s:update({1}, {{'=', -1, 128}})
+---
+- [1, 2, null, 128]
+...
+s:drop()
+---
+...
+-- Negative field number, no field_count
+format = {}
+---
+...
+format[1] = {name = 'field1', type = 'unsigned'}
+---
+...
+format[2] = {name = 'field2', type = 'unsigned'}
+---
+...
+format[3] = {name = 'field3', type = 'unsigned', is_nullable = true}
+---
+...
+format[4] = {name = 'field4', type = 'unsigned', is_nullable = true}
+---
+...
+s = box.schema.create_space('test', {format = format})
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'!', -1, 42}})
+---
+- [1, 2, 42]
+...
+s:update({1}, {{'=', -1, 128}})
+---
+- [1, 2, 128]
+...
+s:drop()
+---
+...
+-- '#' doesn't trim nulls
+s = box.schema.create_space('test')
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'!', 4, 0}})
+---
+- [1, 2, null, 0]
+...
+s:update({1}, {{'#', 4, 1}})
```

Comment:
What about another one 'trim' operation (so that we verify that null is actually stored in the tuple - result should be [1, 2] and not [1, null])?